Dcmi management controller string support

connorgorman opened the pull request.

dougm commented:
Thanks @connorgorman , looks good to me.  I'd prefer to avoid using git submodules however, do you mind removing that part?  Would rather vendor using a tool such as gvt, or even remove the dependency on testify and just using testing from stdlib.

dougm commented:
We should have also seen an automated CLA request from @vmwclabot

connorgorman commented:
@dougm Sounds good. I'll remove the dependency altogether

Commit:
Dcmi management controller string support and dependency management

connorgorman commented:
@dougm I reverted the dependency management piece, but did leave the strechr dependency as I didn't see an easy way to decouple the project from it. Also, the CLA has been sent to Medallia's legal team. Thanks

Commit:
Remove close session printout

Commit:
Fix typo and getMcIdString seg fault

Commit:
added .gitmodules for submodule syncing

Commit:
Revert "added .gitmodules for submodule syncing"

This reverts commit 2f5f539c03e41ffd2e80bfa41626e9d3e0cf4311.

Commit:
Update the console interface to take io.Writer parameters

Commit:
Ensure no overflow if does not end with 0x00

Commit:
Go format the code

Commit:
Fix lanplus response

* Remove newlines received when ipmitool is used

Ezetowers committed:
Merge pull request #3 from medallia/fix_lanplus

Fix lanplus response

Commit:
Add some zeros to raw input to avoid endianess issues

Commit:
Revert previous change and fix non zero issue

Commit:
Improve implementation of GetMcId method

* It now supports multi chunks tags

Commit:
Remove not used import

Ezetowers committed:
Changes in get_mc_id_string implementation (#4)

* Accumulate the amount of bytes read and compare
  it to the amount of bytes to receive. Do not
  ask for more bytes if all bytes have been received

Ezetowers committed:
Fix GetMcId bad parsing (#5)
